aptos-move/e2e-move-tests/tests/rotate_auth_key.rs

```diff
@@ -63,6 +109,88 @@ fn rotate_auth_key_twice() {
     verify_originating_address(&mut harness, account1.auth_key(), *account1.address());
 }
 
+pub fn assert_successful_payload_key_rotation_from_ed25519_to_multi_ed25519(
+    harness: &mut MoveHarness,
+    current_account: Account,
+    originator: AccountAddress,
+    sequence_number: u64,
+    new_private_key: MultiEd25519PrivateKey,
+    new_public_key: MultiEd25519PublicKey,
+) {
+    // construct a proof challenge struct that proves that
+    // the user intends to rotate their auth key
+    let rotation_proof = RotationProofChallenge {
+        account_address: CORE_CODE_ADDRESS,
+        module_name: String::from("account"),
+        struct_name: String::from("RotationProofChallenge"),
+        sequence_number,
+        originator,
+        current_auth_key: AccountAddress::from_bytes(&current_account.auth_key()).unwrap(),
+        new_public_key: new_public_key.to_bytes().to_vec(),
+    };
+
+    let rotation_msg = bcs::to_bytes(&rotation_proof);
+
+    // sign the rotation message by the current private key and the new private key
+    let signature_by_curr_privkey = current_account
+        .privkey
+        .sign_arbitrary_message(&rotation_msg.clone().unwrap());
+    let signature_by_new_privkey = new_private_key.sign_arbitrary_message(&rotation_msg.unwrap());
+
+    assert_success!(harness.run_transaction_payload(
+        &current_account,
+        aptos_stdlib::account_rotate_authentication_key(
+            0,
+            current_account.pubkey.to_bytes().to_vec(),
+            1,
+            new_public_key.to_bytes().to_vec(),
+            signature_by_curr_privkey.to_bytes().to_vec(),
+            signature_by_new_privkey.to_bytes().to_vec(),
+        )
+    ));
+}
+
+pub fn assert_successful_payload_key_rotation_from_ed25519_to_ed25519(
```

Comment:
Maybe not in this PR, but the next one: You can probably use the traits from `aptos-crypto/src/traits.rs` to make this function generic: it is identical to `assert_successful_payload_key_rotation_from_ed25519_to_multi_ed25519` except for the public key types.